src/WizardBlock.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Collection\MapInterface;
+use Dhii\Event\EventFactoryInterface;
+use Dhii\Output\BlockInterface;
+use Dhii\Output\TemplateInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\Modular\Events\EventsConsumerTrait;
+use stdClass;
+
+/**
+ * Class for showing wizard and enqueuing all assets that are related to it.
+ *
+ * @since [*next-version*]
+ */
+class WizardBlock implements BlockInterface
+{
+    /* @since [*next-version*] */
+    use EventsConsumerTrait;
+
+    /**
+     * Is assets attached already.
+     *
+     * @since [*next-version*]
+     *
+     * @var bool
+     */
+    protected static $isAssetsAttached = false;
+
+    /**
+     * Wizard template.
+     *
+     * @since [*next-version*]
+     *
+     * @var TemplateInterface
+     */
+    protected $wizardTemplate;
+
+    /**
+     * Context of wizard template.
+     *
+     * @since [*next-version*]
+     *
+     * @var array|MapInterface|stdClass
+     */
+    protected $context;
+
+    /**
+     * Components templates.
+     *
+     * @since [*next-version*]
+     *
+     * @var string|Stringable
+     */
+    protected $componentsTemplates;
+
+    /**
+     * WizardBlock constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|MapInterface $context             Context of wizard template.
+     * @param TemplateInterface           $wizardTemplate      Wizard template.
+     * @param string|Stringable           $componentsTemplates Components templates.
+     * @param EventManagerInterface       $eventManager        The event manager.
+     * @param EventFactoryInterface       $eventFactory        The event factory.
+     */
+    public function __construct(
+        $context,
```

Comment:
Why is the set of types accepted as context not the same as the set accepted by `TemplateInterface#render()`?